openquake/hazardlib/mfd/arbitrary_mfd.py

```diff
@@ -0,0 +1,98 @@
+# The Hazard Library
+# Copyright (C) 2016, GEM Foundation
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as
+# published by the Free Software Foundation, either version 3 of the
+# License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+"""
+Module :mod:`openquake.hazardlib.mfd.evenly_discretized` defines an evenly
+discretized MFD.
+"""
+from openquake.hazardlib.mfd.base import BaseMFD
+from openquake.baselib.slots import with_slots
+
+
+@with_slots
```

Comment:
We are not using ``__slots__`` anymore, we are using `_slots_` (one underscore instead of two) and that solves all then issues of real ``_slots__``. We could probably even remove the single underscore slots, but that would require thinking and should be done everywhere. It something better left for when we will have no other priorities. For the moment being, it is fine as it is.